deeplake/util/version_control.py

```diff
@@ -471,6 +501,53 @@ def load_version_info(storage: LRUCache) -> Dict:
         )  # backward compatiblity
 
 
+def get_parent_and_reset_commit_ids(version_info, address):
+    """Returns parent commit id and commit id which will be reset. Returns False if address is a non-HEAD commit id"""
+    if address in version_info["branch_commit_map"]:
+        commit_id = version_info["branch_commit_map"][address]
+    elif address in version_info["commit_node_map"]:
+        commit_id = address
+    commit_node = version_info["commit_node_map"][commit_id]
+    if not commit_node.is_head_node:
+        return False
```

Comment:
you're returning only False here, in reset_and_checkout a tuple is expected, this will Fail. Perhaps we should return previous_commit_id, commit_id, is_head to make returns consistent as well